vue/test/components/dropdown/dropdown.test.js

```diff
@@ -0,0 +1,165 @@
+        });
+
+        const item1Index = 0;
+        const item1 = component.findAll(".dropdown-item").at(item1Index);
+        await item1.trigger("mouseenter");
+        await item1.trigger("mouseover");
+        assert.strictEqual(component.vm.$data.highlightedData[item1Index], true);
+        await item1.trigger("mouseleave");
+        assert.strictEqual(component.vm.$data.highlightedData[item1Index], undefined);
+
+        const item2Index = 1;
+        const item2 = component.findAll(".dropdown-item").at(item2Index);
+        await item2.trigger("mouseenter");
+        await item2.trigger("mouseover");
+        assert.strictEqual(component.vm.$data.highlightedData[item2Index], true);
+        await item2.trigger("mouseleave");
+        assert.strictEqual(component.vm.$data.highlightedData[item2Index], undefined);
+    });
+
+    it("should send the correct highlight event", async () => {
+        const items = [
+            {
+                value: "text_1",
+                label: "Text 1"
+            },
+            {
+                value: "text_2"
+            }
+        ];
+        const component = base.getComponent("Dropdown", {
+            props: { items: items }
+        });
+
+        const itemIndex1 = 0;
+        const item1 = component.findAll(".dropdown-item").at(itemIndex1);
+        await item1.trigger("mouseenter");
+        await item1.trigger("mouseover");
+        let highlightedItem = parseInt(
+            Object.keys(component.emitted("update:highlighted")[0][0])[0]
+        );
+        assert.strictEqual(component.emitted("update:highlighted").length, 1);
+        assert.strictEqual(highlightedItem, highlightedItem);
+        await item1.trigger("mouseleave");
+
+        const itemIndex2 = 1;
+        const item2 = component.findAll(".dropdown-item").at(itemIndex2);
+        await item2.trigger("mouseenter");
+        await item2.trigger("mouseover");
+        highlightedItem = parseInt(Object.keys(component.emitted("update:highlighted")[2][0])[0]);
+        assert.strictEqual(component.emitted("update:highlighted").length, 3);
+        assert.strictEqual(highlightedItem, itemIndex2);
+    });
+
+    it("should send the correct click event", async () => {
+        const items = [
+            {
+                value: "text_1",
+                label: "Text 1"
+            },
+            {
+                value: "text_2"
+            }
+        ];
+        const component = base.getComponent("Dropdown", {
+            props: { items: items }
+        });
+
+        const item = component.findAll(".dropdown-item").at(1);
+        await item.trigger("click");
+        const clickedItemValue = component.emitted("item-clicked")[0][0].value;
+        const clickedItemIndex = component.emitted("item-clicked")[0][1];
+        assert.strictEqual(component.emitted("item-clicked").length, 1);
+        assert.strictEqual(item.text(), clickedItemValue);
+        assert.strictEqual(1, clickedItemIndex);

```

Comment:
Set a variable with the result of `component.emitted("item-clicked")` to reduce the `component.emitted` calls, like this:
`const clickedEvents = component.emitted("item-clicked");`

Also, in the asserts, do something like this:
`assert.deepEqual(clickedEvents[0], [{ value: 'text_2' }, 1]);`